pkg/trace/agent/agent.go

```diff
@@ -255,12 +250,12 @@ func (a *Agent) Process(p *api.Payload, sublayerCalculator *stats.SublayerCalcul
 			ss.Size += pb.Trace(events).Msgsize()
 		}
 		if ss.Size > writer.MaxPayloadSize {
-			a.Out <- ss
+			a.TraceWriter.In <- ss
 			ss = new(writer.SampledSpans)
 		}
 	}
 	if ss.Size > 0 {
-		a.Out <- ss
+		a.TraceWriter.In <- ss
```

Comment:
Would it be a better API to instead have a method returning this channel as `chan<- *SampledSpans`? That would make sure nobody outside the trace writer can receive from it. (e.g. `a.TraceWriter.In()`)